FedoraApi.php

```diff
@@ -346,6 +346,38 @@ public function getDatastreamDissemination($pid, $dsid, $as_of_date_time = NULL,
     return $response;
   }
 
+  /**
+   * Get the datastream profile.
+   *
+   * @param String $pid
+   *   Persistent identifier of the digital object.
+   * @param String $dsid
+   *   Datastream identifier.
+   * @param String|null $as_of_date_time
+   *   (optional) Indicates that the result should be relative to the
+   *     digital object as it existed at the given date and time. Defaults to
+   *     the most recent version.
+   *
+   * @throws RepositoryException
+   *
+   * @return array
+   *   The parsed datastream profile.
+   */
+  public function getDatastreamProfile($pid, $dsid, $as_of_date_time = NULL) {
+    $pid = urlencode($pid);
+    $dsid = urlencode($dsid);
+    $separator = '?';
+
+    $request = "/objects/$pid/datastreams/$dsid";
+    $this->connection->addParam($request,$separator, 'asOfDateTime', $as_of_date_time);
```

Comment:
So, the connection parameter "asOfDateTime" is mandatory even if NULL?

Reply:
Ok, just read the API docs at https://wiki.duraspace.org/display/FEDORA38/REST+API#RESTAPI-getDatastream and yes, if NULL then its like a normal getDatastream, which leads to, would it not be better to add the asOfDateTime option to getDatastream as an extra argument? instead of a new method? Thinking about use case of /at certain point/+ checksum

Reply:
addParam ignores any NULL values so it is not mandatory.

Reply:
OK, if it gets ignored when NULL then for NULL it behaves like a normal getDatastream. Would make sense just call getDatastream in case of NULL to avoid duplicated code? Duno, but i still prefer the idea of extending  `::getDatastream()` by adding a new argument instead of creating a new method

Reply:
Oh this is me being confused by API-A vs M, I didn't realize this function was part of the 'M' functions. Yes you are right I will add the parameter there.

The A vs M distinction doesn't seem to be something that Tuque should have to force the Drupal developer to think about IMO.